### PR TITLE
FEATURE: Add support for multiple schedule queues.

### PR DESCRIPTION
**Pull request by udan11**

### CODE REVIEW

**Comment by SamSaffron** [comment]
Perfect, this feels so much tidier and flexible, be sure to update the readme here with some documentation and merge away!

**Comment by SamSaffron** [comment]
Once you feel confident here, feel free to merge

File: lib/mini_scheduler.rb [hunk 1]
```diff
@@ -52,24 +52,28 @@ def self.skip_schedule(&blk)
...
 
   def self.start
-    manager = Manager.new
+    Manager.discover_queues.each do |queue|
+      manager = Manager.new(queue: queue)
 
-    Manager.discover_schedules.each do |schedule|
-      manager.ensure_schedule!(schedule)
-    end
+      Manager.discover_schedules.each do |schedule|
```
Comment (ZogStriP) [inline lib/mini_scheduler.rb]: Small perf remark: should we move that outside the loop and re-use it? Something like

``` ruby
schedules = Manager.discover_schedules

Manager.discover_queues.each do |queue|
  schedules.each do |schedule|
    manager.ensure_schedule!(schedule) if schedule.queue == queue
  end
end
```

File: lib/mini_scheduler/manager.rb [hunk 2]
```diff
@@ -293,11 +291,19 @@ def keep_alive
     end
 
     def lock
-      MiniScheduler::DistributedMutex.synchronize(Manager.lock_key, MiniScheduler.redis) do
+      MiniScheduler::DistributedMutex.synchronize(Manager.lock_key(queue), MiniScheduler.redis) do
         yield
       end
     end
 
+    def self.discover_queues
```
Comment (ZogStriP) [inline lib/mini_scheduler/manager.rb]: Would that work?

```suggestion
    def self.discover_queues
      ObjectSpace.each_object(MiniScheduler::Schedule).map(&:queue).to_set
    end
```

Comment (udan11) [reply lib/mini_scheduler/manager.rb]: Yes, it does!

**Pull request merged**